packages/host/app/components/card-catalog/index.gts

```diff
@@ -1,103 +1,122 @@
-                  />
-                </li>
-              {{/let}}
-            {{/each}}
-          </ul>
-          {{#if (gt realm.cards.length realm.displayedCards.length)}}
-            <Button
-              {{on 'click' (fn this.displayMoreCards realm)}}
-              @kind='secondary-light'
-              @size='small'
-              data-test-show-more-cards
+      {{#if @realmInfos}}
+        {{#let
+          (this.groupByRealmUrl @cards this.pageSize)
+          as |groupedCardsByRealm|
+        }}
+          {{#each-in groupedCardsByRealm as |_realmUrl realmData|}}
+            <section
+              class='card-catalog__realm'
+              data-test-realm={{realmData.realmInfo.name}}
             >
-              Show more cards
-            </Button>
-          {{/if}}
-        </section>
-      {{else}}
-        <p>No cards available</p>
-      {{/each}}
+              <CardCatalogResultsHeader
+                @realm={{realmData.realmInfo}}
+                @resultsCount={{realmData.cardsTotal}}
+              />
+
+              <ul class='card-catalog__group'>
+                {{#each realmData.cards as |card index|}}
+                  {{#if (lt index realmData.displayedCardsCount)}}
+                    {{#let (eq @selectedCardUrl card.url) as |isSelected|}}
+                      <li class={{cn 'item' selected=isSelected}}>
+                        <button
+                          class='catalog-item {{if isSelected "selected"}}'
+                          {{on 'click' (fn @select card.url)}}
+                          {{on 'dblclick' (fn @select card.url)}}
+                          {{on 'keydown' (fn this.handleEnterKey card.url)}}
+                          data-test-select={{removeFileExtension card.url}}
+                          aria-label='Select'
+                          data-test-card-catalog-item={{removeFileExtension
+                            card.url
+                          }}
+                          data-test-card-catalog-item-selected={{isSelected}}
+                        >
+                          {{card.component}}
+                        </button>
+                      </li>
+                    {{/let}}
+                  {{/if}}
+                {{/each}}
+              </ul>
+
+              {{#if (gt realmData.cardsTotal realmData.displayedCardsCount)}}
+                <Button
+                  {{on
+                    'click'
+                    (fn
+                      (mut realmData.displayedCardsCount)
+                      (add realmData.displayedCardsCount this.pageSize)
+                    )
+                  }}
+                  @kind='secondary-light'
+                  @size='small'
+                  data-test-show-more-cards
+                >
+                  Show
+                  {{this.pageSize}}
+                  more cards ({{subtract
+                    realmData.cardsTotal
+                    realmData.displayedCardsCount
```

Comment:
there's nothing wrong with this approach, but there is a school of thought that computeds `get`) are easier to read in templates. putting lots of function calls into a template can make it harder to read. it's more of a style issue

Reply:
I see, simplified the template, it's now easier to read